src/derivation.test.ts

```diff
@@ -63,105 +67,122 @@ describe('derivation', () => {
-        deriveKeyFromPath([
+        return deriveKeyFromPath([
           bip39Part,
           bip32Part1.replace(`'`, '"') as any,
           ...rest,
         ]);
-      }).toThrow(/Invalid HD path segment: The path segment is malformed\./u);
+      }).rejects.toThrow(
+        /Invalid HD path segment: The path segment is malformed\./u,
+      );
 
-      expect(() => {
+      await expect(() =>
         deriveKeyFromPath(
           [bip39Part, ethereumBip32PathParts[0]],
           Buffer.alloc(64).fill(1),
           0,
-        );
-      }).toThrow(
+        ),
+      ).rejects.toThrow(
         /Invalid HD path segment: The segment must consist of a single BIP-39 node for depths of 0\. Received:/u,
       );
 
       // bip39 seed phrase component must be completely lowercase
-      expect(() => {
-        deriveKeyFromPath([bip39Part.replace('r', 'R') as any]);
-      }).toThrow(/Invalid HD path segment: The path segment is malformed\./u);
+      await expect(() =>
+        deriveKeyFromPath([bip39Part.replace('r', 'R') as any]),
+      ).rejects.toThrow(
+        /Invalid HD path segment: The path segment is malformed\./u,
+      );
 
       // Multipaths that start with bip39 segment require _no_ parentKey
-      expect(() => {
-        deriveKeyFromPath([bip39Part], parentKey);
-      }).toThrow(
+      await expect(() =>
+        deriveKeyFromPath([bip39Part], parentKey),
+      ).rejects.toThrow(
         /Invalid derivation parameters: May not specify parent key if the path segment starts with a BIP-39 node\./u,
       );
 
       // Multipaths that start with bip32 segment require parentKey
-      expect(() => {
-        deriveKeyFromPath([`bip32:1'`]);
-      }).toThrow(
+      await expect(() => deriveKeyFromPath([`bip32:1'`])).rejects.toThrow(
         /Invalid derivation parameters: Must specify parent key if the first node of the path segment is not a BIP-39 node\./u,
       );
 
       // parentKey must be a buffer if specified
-      expect(() => {
-        deriveKeyFromPath([`bip32:1'`], parentKey.toString('base64') as any);
-      }).toThrow('Parent key must be a Buffer if specified.');
+      await expect(() =>
+        deriveKeyFromPath([`bip32:1'`], parentKey.toString('base64') as any),
+      ).rejects.toThrow('Parent key must be a Buffer if specified.');
     });
   });
 
   describe('bip32Derive', () => {
-    it('derives the expected keys and addresses', () => {
+    it('derives the expected keys and addresses', async () => {
       // generate parent key
       let parentKey: Buffer;
-      parentKey = bip39Derive(mnemonic);
-      parentKey = bip32Derive(`44'`, parentKey);
-      parentKey = bip32Derive(`60'`, parentKey);
-      parentKey = bip32Derive(`0'`, parentKey);
-      parentKey = bip32Derive(`0`, parentKey);
-      const keys = expectedAddresses.map((_, index) => {
-        return bip32Derive(`${index}`, parentKey);
-      });
+
+      /* eslint-disable require-atomic-updates */
```

Comment:
Why does eslint not like this? 🤔

Reply:
and is it safe to disable 😅

Reply:
```
ESLint: Possible race condition: `parentKey` might be reassigned based on an outdated value of `parentKey`.(require-atomic-updates)
```

Reply:
look ok to me, nothing changes `parentKey` in the meantime

Reply:
Yeah, I think it's fine. ESLint is trying to protect us from interweaving async operations and the `parentKey` changing in between our async function calls here. I don't see how that could happen here, so we should be good.